pandas/plotting/_core.py

```diff
@@ -574,6 +576,13 @@ def _get_xticks(self, convert_period=False):
 
     @classmethod
     def _plot(cls, ax, x, y, style=None, is_errorbar=False, **kwds):
+        # GH25587: cast ExtensionArray of pandas (IntegerArray, etc.) to
```

Comment:
Thoughts on moving this conversion to `_compute_plot_data`? It may be more appropriate there.

Reply:
In `_compute_plot_data` we only get the DataFrame, some columns in data frame may be `nd.array`, some may be `IntegerArray`. Moving conversion to here means we need traverse the data frame and do conversion on every column if necessary and reconstruct a DataFrame. I think that may hurt performance.

Reply:
Scanning the dataframe for extension arrays isn't that expensive, relative to the cost of converting and plotting.

My concern is that we're missing the conversion of EA to ndarray in other plot types (scatter, area, etc). Can you add tests for those to see if they fail?

Reply:
After move the conversion to `_compute_plot_data`, `kde` and `area` plots still don't work because we cast `IntegerArray` to `nd.array(dtype=object)`, but `kde` and `area` needs dtype that can be used to ufunc `add` of numpy. Could we ignore the tests for `kde` and `area` and open a new issue to track that trouble?

<details><summary>The traceback of `kde`</summary>

```
  File "D:\Open\pandas\pandas\plotting\_core.py", line 1474, in _plot
    gkde = gaussian_kde(y, bw_method=bw_method)
  File "D:\Program\Conda\lib\site-packages\scipy\stats\kde.py", line 208, in __init__
    self.set_bandwidth(bw_method=bw_method)
  File "D:\Program\Conda\lib\site-packages\scipy\stats\kde.py", line 540, in set_bandwidth
    self._compute_covariance()
  File "D:\Program\Conda\lib\site-packages\scipy\stats\kde.py", line 551, in _compute_covariance
    aweights=self.weights))
  File "D:\Program\Conda\lib\site-packages\numpy\lib\function_base.py", line 2427, in cov
    avg, w_sum = average(X, axis=1, weights=w, returned=True)
  File "D:\Program\Conda\lib\site-packages\numpy\lib\function_base.py", line 419, in average
    scl = wgt.sum(axis=axis, dtype=result_dtype)
  File "D:\Program\Conda\lib\site-packages\numpy\core\_methods.py", line 36, in _sum
    return umr_sum(a, axis, dtype, out, keepdims, initial)
TypeError: No loop matching the specified signature and casting was found for ufunc add
```
</details>

<details><summary>The traceback of `area`</summary>

```
  File "D:\Open\pandas\pandas\plotting\_core.py", line 252, in generate
    self._make_plot()
  File "D:\Open\pandas\pandas\plotting\_core.py", line 1000, in _make_plot
    **kwds)
  File "D:\Open\pandas\pandas\plotting\_core.py", line 1151, in _plot
    cls._update_stacker(ax, stacking_id, y)
  File "D:\Open\pandas\pandas\plotting\_core.py", line 1083, in _update_stacker
    ax._stacker_pos_prior[stacking_id] += values
TypeError: ufunc 'add' output (typecode 'O') could not be coerced to provided output parameter (typecode 'd') according to the casting rule ''same_kind''
```
</details>